sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -27,6 +27,14 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     uint8_t type = (flags & IEEE802154_FCF_TYPE_MASK);
     uint8_t bcast = (flags & IEEE802154_BCAST);
 
+    if (dst_pan.u16 == src_pan.u16 &&
+        (src_len != 0 && dst_len != 0)) {
```

Comment:
> @miri64 I don't understand why src address and length are set in _send here but the destination is not set the same way when it is multicast? I mean the code in gnrc_netdev2_ieee802154.c already knows about L2 so it should be able to set the address like here, or not?

Because the driver is not "supposed" to know the length of the broadcast address. That's this function's job.

> and what's the reasoning behind unset the broadcast flag here?

The flag is not part of the IEEE 802.15.4 specs so adding it to the MAC header seems like a bad idea ;-).